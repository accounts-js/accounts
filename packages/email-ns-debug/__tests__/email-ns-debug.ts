import EmailNSDebug from '../src';

const passwordEnrollSpy = jest.fn(()=>true)

const passwordPlugin = {
  name: 'password',
  enroll: passwordEnrollSpy
}

const config = {
  plugins: [passwordPlugin]
}

const emailNSDebug = new EmailNSDebug(config)

describe('EmailNSMailgun', () => {


  describe('send',() => {

    it('should print the email in console', () => {
      const temp = console.dir;
      console.dir = jest.fn(()=>true);
      emailNSDebug.send('test');
      expect(console.dir).toBeCalled();
      console.dir = temp;
    })
    
  })


  describe('notify',() => {
    

    it('should throw if plugin not found', () => {
      expect(()=>emailNSDebug.notify('oauth')).toThrowErrorMatchingSnapshot()
    })

    it('should throw if action not found on plugin', () => {
      expect(()=>emailNSDebug.notify('password', 'action')).toThrowErrorMatchingSnapshot()
    })

    it('should call the action if found on plugin', () => {
      emailNSDebug.notify('password', 'enroll', { test: true })
      expect(passwordEnrollSpy).toBeCalled()
    })
    
  })

})
