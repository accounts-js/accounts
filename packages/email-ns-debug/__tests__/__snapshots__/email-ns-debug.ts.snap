// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailNSMailgun notify should throw if action not found on plugin 1`] = `"[ Accounts - EmailNotificationServiceDebug ] notify : The action \\"action\\" is not found on plugin \\"password\\""`;

exports[`EmailNSMailgun notify should throw if plugin not found 1`] = `"[ Accounts - EmailNotificationServiceDebug ] notify : The plugin \\"oauth\\" is not registered"`;
